x/dex/keeper/pool.go

```diff
@@ -136,19 +136,20 @@ func (k Keeper) GetPoolIDByParams(
 }
 
 func (k Keeper) SetPool(ctx sdk.Context, pool *types.Pool) {
-	k.updatePoolReserves(ctx, pool.LowerTick0)
-	k.updatePoolReserves(ctx, pool.UpperTick1)
+	k.updatePoolReserves(ctx, pool.LowerTick0, pool.GetId())
+	k.updatePoolReserves(ctx, pool.UpperTick1, pool.GetId())
 
 	// TODO: this will create a bit of extra noise since not every Save is updating both ticks
 	// This should be solved upstream by better tracking of dirty ticks
 	ctx.EventManager().EmitEvent(types.CreateTickUpdatePoolReserves(*pool.LowerTick0))
 	ctx.EventManager().EmitEvent(types.CreateTickUpdatePoolReserves(*pool.UpperTick1))
 }
 
-func (k Keeper) updatePoolReserves(ctx sdk.Context, reserves *types.PoolReserves) {
+func (k Keeper) updatePoolReserves(ctx sdk.Context, reserves *types.PoolReserves, poolID uint64) {
 	if reserves.HasToken() {
 		k.SetPoolReserves(ctx, reserves)
 	} else {
+		ctx.EventManager().EmitEvents(types.GetEventsDecTotalPoolReserves(poolID))
```

Comment:
I would use pairID not poolID for consistency

Reply:
PairId does not need to be passed all the way through. It can be accessed via `reserves.Key.TradePairId.MustPairID()`